### PR TITLE
fix(web): Enhance exception handling to return detailed message for the standard exceptions

### PR DESCRIPTION
**Pull request by Aloren**
- Enhance exception handling to return detailed message for the standard exceptions (like invalid requests) handled by Spring (see: ResponseEntityExceptionHandler). 
- Enhance controller method definitions.

Motivation behind this change: when user sends invalid request or any other internal error occurs, he/she gets only http status in response and no reason for what is the issue; this pull request addresses this issue and adds simple json response body with `message` field.

### CODE REVIEW

File: kayenta-web/src/main/java/com/netflix/kayenta/controllers/ControllerExceptionHandler.java [hunk 1]
```diff
@@ -19,13 +19,17 @@
...
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestControllerAdvice;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
 
 @RestControllerAdvice
-public class ControllerExceptionHandler {
+public class ControllerExceptionHandler extends ResponseEntityExceptionHandler {
```
Comment (Aloren) [inline kayenta-web/src/main/java/com/netflix/kayenta/controllers/ControllerExceptionHandler.java]: https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/servlet/mvc/method/annotation/ResponseEntityExceptionHandler.html

**Comment by fieldju** [comment]
I am all about more context in the returned errors: see https://github.com/spinnaker/kayenta/issues/354

Are the changes you are making backwards compatible with Kayentas Deck and Orca integrations?

What are the before and after shapes of the error responses?

**Comment by fieldju** [comment]
Does this sort of logic belong in https://github.com/spinnaker/kork/blob/master/kork-web/src/main/java/com/netflix/spinnaker/kork/web/exceptions/GenericExceptionHandlers.java?

Im not 100% sure what the pattern is supposed to be with Kork vs implementing this in the individual Spinnaker services. It seems like the exception handling logic should be consistent across the whole Spinnaker ecosystem.

Maybe @cfieber or @robzienert could chim in here on Spinnaker best practices?

**Comment by Aloren** [comment]
> Does this sort of logic belong in https://github.com/spinnaker/kork/blob/master/kork-web/src/main/java/com/netflix/spinnaker/kork/web/exceptions/GenericExceptionHandlers.java?

Thank you. I was looking initially for such sort of thing, but somehow missed it. I'm not sure, but it looks to me that this is actually not used in Kayenta. Will check that now.

**Comment by Aloren** [comment]
Okay, so had a look into /beans endpoint and found:
```
"genericExceptionHandlers": {
"aliases": [],
"scope": "singleton",
"type": "com.netflix.spinnaker.kork.web.exceptions.GenericExceptionHandlers",
"resource": "class path resource [com/netflix/spinnaker/config/ErrorConfiguration.class]",
"dependencies": []
},
```
so this is actually setup.

I commented out the newly added behavior in `ControllerExceptionHandler` and added `ErrorConfiguration` into the tests setup. Now 
`storeCanaryConfig_returnsBadRequestResponseForMissingRequestBody` fails with:
```
MockHttpServletResponse:
           Status = 500
    Error message = Required request body is missing: public com.netflix.kayenta.canary.CanaryConfigUpdateResponse com.netflix.kayenta.controllers.CanaryConfigController.storeCanaryConfig(java.lang.String,com.netflix.kayenta.canary.CanaryConfig)
          Headers = [X-Content-Type-Options:"nosniff", X-XSS-Protection:"1; mode=block", Cache-Control:"no-cache, no-store, max-age=0, must-revalidate", Pragma:"no-cache", Expires:"0", X-Frame-Options:"DENY"]
     Content type = null
             Body = 
    Forwarded URL = null
   Redirected URL = null
          Cookies = []
```

**Comment by Aloren** [comment]
I suppose that `extends ResponseEntityExceptionHandler` should be added into `GenericExceptionHandlers` (`kork-web`), so that all spring-mvc exceptions were processed correctly across all spinnaker web applications.

**Comment by robzienert** [comment]
1. Thank you for cleaning up the `RequestMapping` annots.
1. I'd prefer to see improvements around our RPC exceptions to be applied across all projects. So, I'd prefer to see those changes land in kork if at all possible.

**Comment by robzienert** [comment]
Ping. I like the contribution but I think it should be more broadly applied across the Spinnaker services.

**Comment by Aloren** [comment]
@robzienert I will move implementatio to kork-web module then and leave here only endpoints enhancements. Thanks

**Comment by Aloren** [comment]
closing in favor of: 
https://github.com/spinnaker/kayenta/pull/681
https://github.com/spinnaker/kork/pull/535

**Pull request closed without merging**